src/event.rs

```diff
@@ -1,4 +1,18 @@
-//! This module contains the `WindowEvent` type and re-exports glutin event
+//! This module contains the `Event` enum and re-exports glutin event
 //! types.
 
-pub use winit::{ElementState, Event, KeyboardInput, MouseButton, VirtualKeyCode, WindowEvent};
+pub use winit::{ElementState, Event as WinitEvent, KeyboardInput, MouseButton, VirtualKeyCode,
+                WindowEvent};
+
+/// Events that the engine might send to the user.
+pub enum Event {
+    /// An event from the Winit crate
+    WinitEvent(WinitEvent),
+    /// Unused event, enables _ in match statements
+    UnusedEvent,
```

Comment:
I don't get why we need this variant.

Reply:
As I understand it there is only one variant for now. And @Xaeroxe use `_` arm in matches so those matches will stay valid when new variants will be added here.

Reply:
Although I was thinking `_` arm could be added even if match have arm for all variants.

Reply:
Well. I've checked. It is possible but causes warning.

Reply:
Well we should make this enum future proof and add a `__NonExhaustive` variant instead.

Reply:
Huh, I actually didn't know about `__NonExhaustive` thanks!

Reply:
It does not matter what it is called as it is not a standard feature. `__Nonexhaustive` is a std-only hack. The best we can do at the moment is a `#[doc(hidden)]` variant.

[#[non_exhaustive] is on its way](https://github.com/rust-lang/rust/issues/44109)

Reply:
@stenvorbis Yeah I don't really care how it's called, but prefixing with `__` makes sense and this name is somehow a known pattern. Ye, we should have `#[doc(hidden)]`